File: crates/versionlens-vscode-model/src/diagnostic.rs
use serde::{Deserialize, Serialize};

use crate::Range;

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct DiagnosticPayload {
    pub range: Range,
    pub message: String,
    pub severity: u8,
    pub source: Option<String>,
    pub code: Option<String>,
    pub code_description_url: Option<String>,
}

File: packages/vscode-extension/src/extension/__tests__/diagnostics/constructors/diagnostic.ts
class DiagnosticMock {
  message: string;
  range: unknown;
  severity: number;

  constructor(range: unknown, message: string, severity: number) {
    this.range = range;
    this.message = message;
    this.severity = severity;
  }
}

export { DiagnosticMock };
